src/github.com/getlantern/errors/errors.go

```diff
@@ -0,0 +1,641 @@
+/*
+Package errors defines error types used across Lantern project.
```

Comment:
So here's the new API. It seems more expressive and requires less keystoke. @oxtoacart @uaalto any thoughts?

Instead of changing `golog` package, I make `errors` package to call `golog.Error()` if logging flag is set globally. With the `stack` package, we can attach the call stack when error is created, and attach another call stack when error is reported, if necessary.